strawberryfields/api/result.py

```diff
@@ -118,7 +118,15 @@ def state(self):
 
     def __repr__(self):
         """String representation."""
-        shots, modes = self.samples.shape
+        try:
+            shots, modes = self.samples.shape
+        except ValueError:
+            # if the samples has dim 3, then they're from a TDMProgram
+            shots, modes, timebins = self.samples.shape
+            return "<Result: spatial_modes={}, shots={}, timebins={} contains state={}>".format(
+                modes, shots, timebins, self._is_stateful
```

Comment:
Think this would be good to keep the order from the shape. Let me know if the change in order was deliberate though!
```suggestion
            return "<Result: shots={}, spatial_modes={}, timebins={} contains state={}>".format(
                shots, modes, timebins, self._is_stateful
```

Reply:
I also think this looks nicer to be honest. I just kept the same form as the current repr (for GBS samples) which starts with modes, and then shots. I might just update it in both places if you also think that makes more sense.

Reply:
Hmm, think it's good to keep the shape order. As for changing for GBS, not quite sure if that would cause any troubles somewhere (don't think so) :thinking:.